datalad/support/repo.py

```diff
@@ -249,3 +249,32 @@ class RepoInterface(object):
 
     # Test!
     pass
+
+
+def path_based_str_repo(cls):
+    """A helper decorator for a class to define str and repo based on its .path
```

Comment:
s/repo/repr/ in several places.

Other minor issues I noticed with the text on a quick scan: first sentence does not parse, "cut" is probably a confusing word to use, double space between "idea" and "is".

Reply:
Cool, thanks for picking those up!... I will tune up those if we decide that is a way to go (or welcome to push fixup to make it to your liking)